assets/js/utils/rtcUtils.ts

```diff
@@ -184,6 +181,24 @@ export const connectWebrtc = async (
   return webrtc;
 };
 
+export const connectPresentersTracks = async (
+  playerCallbacks: { [key: string]: (sourceType: SourceType) => void },
+  setPresenters: React.Dispatch<React.SetStateAction<{ [key: string]: Presenter }>>
+) => {
+  setPresenters((presenters) => {
+    Object.values(presenters)
+      .filter(
+        (presenter) => presenter.status == "connecting" && presenter.connectCallbacks.length > 0
+      )
+      .forEach((presenter) => {
+        const playerCallback = playerCallbacks[presenter.email];
+        presenter.connectCallbacks.forEach((callback) => callback(playerCallback));
+        presenters = { ...presenters, [presenter.email]: { ...presenter, connectCallbacks: [] } };
+      });
+    return presenters;
+  });
```

Comment:
Could we move this to some function like `doPresenterCallback` or something like this?